src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/DefaultComesLastCheck.java

```diff
@@ -71,15 +74,22 @@ public void visitToken(DetailAST ast) {
         final DetailAST defaultGroupAST = ast.getParent();
         //default keywords used in annotations too - not what we're
         //interested in
+        final Set<Integer> tokenTypes = new HashSet<>();
+        if (ast.getPreviousSibling() != null) {
+            tokenTypes.add(ast.getPreviousSibling().getType());
+        }
+        if (ast.getNextSibling() != null) {
+            tokenTypes.add(ast.getNextSibling().getType());
+        }
         if (defaultGroupAST.getType() != TokenTypes.ANNOTATION_FIELD_DEF
                 && defaultGroupAST.getType() != TokenTypes.MODIFIERS) {
             final DetailAST switchAST = defaultGroupAST.getParent();
             final DetailAST lastGroupAST =
                 switchAST.getLastChild().getPreviousSibling();
-
-            if (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
+            if (!tokenTypes.contains(TokenTypes.LITERAL_CASE)
+                && (defaultGroupAST.getLineNo() != lastGroupAST.getLineNo()
                 || defaultGroupAST.getColumnNo()
```

Comment:
indentation here should increase since we are now wrapping on the `(`. It shouldn't remain on same level as `&&`.